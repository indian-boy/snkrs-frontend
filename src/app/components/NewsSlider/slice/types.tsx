import { FunctionComponent, SVGProps } from 'react';

export interface SliderItem {
  id: number;
  title: string;
  link: {
    label: string;
    url: string;
  };
  Image: FunctionComponent<
    SVGProps<SVGSVGElement> & { title?: string | undefined }
  >;
}

export interface NewsSliceState {
  selected: SliderItem;
  items: SliderItem[];
}
