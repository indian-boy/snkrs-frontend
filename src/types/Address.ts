export interface Address {
  latitude: number;
  longitude: number;
  location: string;
}
