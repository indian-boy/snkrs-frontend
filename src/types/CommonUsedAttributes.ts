export interface CommonUsedAttributes {
  'data-testid': string;
}
