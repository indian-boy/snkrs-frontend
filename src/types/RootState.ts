import { NewsSliceState } from 'app/components/NewsSlider/slice/types';
import { ThemeState } from 'styles/themes/slice/types';

// [IMPORT NEW CONTAINERSTATE ABOVE] < Needed for generating containers seamlessly

/*
  Because the redux-injectors injects your reducers asynchronously somewhere in your code
  You have to declare them here manually
*/
export interface RootState {
  theme?: ThemeState;
  newsSlice?: NewsSliceState;
  // [INSERT NEW REDUCER KEY ABOVE] < Needed for generating containers seamlessly
}
