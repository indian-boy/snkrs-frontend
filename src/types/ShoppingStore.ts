import { Address } from './Address';

export interface ShoppingStore {
  id: number;
  placeName: string;
  address: Address;
  availability: number;
  serviceHours: string;
}
