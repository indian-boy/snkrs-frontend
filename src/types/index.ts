import { Address } from './Address';
import { RootState } from './RootState';
import { ShoppingStore } from './ShoppingStore';

export type { RootState, Address, ShoppingStore };
