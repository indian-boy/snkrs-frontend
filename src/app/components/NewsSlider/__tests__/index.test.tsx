import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { CommonUsedAttributes } from 'types';
import { NewsSlider } from '..';

const renderComponentWithProviders = (
  props: Parameters<typeof NewsSlider>[number] & CommonUsedAttributes,
  store: Store,
) =>
  render(
    <Provider store={store}>
      <NewsSlider {...props} />
    </Provider>,
  );

describe('<NewsSlider  />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const { getByTestId, container } = renderComponentWithProviders(
      {
        'data-testid': 'newsSliderID',
      },
      store,
    );
    const newsSliderElement = getByTestId('newsSliderID');

    expect(newsSliderElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
