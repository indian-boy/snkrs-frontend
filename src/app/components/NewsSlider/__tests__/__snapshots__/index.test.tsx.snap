// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsSlider  /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F7F7F7;
  padding: 0.5rem 0;
}

.c2 {
  width: 1rem;
  height: 1rem;
}

.c6 {
  width: 1rem;
  height: 1rem;
}

.c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #1A1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.5rem;
  max-width: 40rem;
  text-align: center;
  -webkit-animation: fQRpvB 0.6s ease-in-out;
  animation: fQRpvB 0.6s ease-in-out;
}

.c5 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A1A1A;
  font-size: 0.875rem;
  line-height: 1.5rem;
  white-space: nowrap;
}

.c1 {
  background-color: #FFFFFF;
  padding: 0.5rem;
  border-radius: 50%;
  font-size: 0;
  cursor: pointer;
  outline: none;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
}

@media (min-width:1024px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    gap: 1.5rem;
  }
}

@media (min-width:1024px) {
  .c4 {
    gap: 1.5rem;
  }
}

<div
  class="c0"
  data-testid="newsSliderID"
>
  <button
    class="c1"
  >
    <svg
      class="c2"
    >
      arrow_left.svg
    </svg>
  </button>
  <div
    class="c3"
  >
    <svg>
      nike_snkrs.svg
    </svg>
    <div
      class="c4"
    >
      Fique por dentro dos lan√ßamentos
      <a
        aria-current="page"
        class="c5 active"
        href="/"
      >
        SKNRS
      </a>
    </div>
  </div>
  <button
    class="c1"
  >
    <svg
      class="c6"
    >
      arrow_right.svg
    </svg>
  </button>
</div>
`;
